src/main/java/gov/hhs/cms/bluebutton/fhirstress/backend/CustomSamplerClient.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.jmeter.protocol.java.sampler.JavaSamplerContext;
+import org.apache.jmeter.samplers.SampleResult;
+
+//import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.client.IGenericClient;
+import ca.uhn.fhir.rest.client.exceptions.FhirClientConnectionException;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+//import gov.hhs.cms.bluebutton.data.model.rif.RifFileType;
+import gov.hhs.cms.bluebutton.fhirstress.utils.FhirClient;
+import gov.hhs.cms.bluebutton.fhirstress.utils.RifParser;
+
+/**
+ * {@link CustomSamplerClient} encapsulates the common variables and methods
+ * that are shared among all tests in this package. Simply derive from this
+ * class and implement {@link CustomSamplerClient#executeTest()} at a minimum to
+ * create new test functionality.
+ */
+public abstract class CustomSamplerClient extends AbstractJavaSamplerClient {
+	protected static final String PARAM_SERVER = "fhir_server";
+	protected static final String KEYSTORE_DIR = "keystore_dir";
+	protected static final String PROXY_HOST = "proxy_host";
+	protected static final String PROXY_PORT = "proxy_port";
+	protected static final String RIFFILE = "riffile";
+	protected static final String RIFTYPE = "riftype";
+	protected static final String HOSTNAME_UNKNOWN = "unknown-host";
+
+	/**
+	 * Abstract method that should be implemented by derived classes to implement
+	 * test functionality specific to the desired test case.
+	 */
+	protected abstract void executeTest();
+
+	protected String hostName;
+	protected IGenericClient client;
+	protected RifParser parser = null;
+
+	/**
+	 * @see org.apache.jmeter.protocol.java.sampler.AbstractJavaSamplerClient#getDefaultParameters()
+	 */
+	@Override
+	public Arguments getDefaultParameters() {
+		Arguments defaultParameters = new Arguments();
+		defaultParameters.addArgument(PARAM_SERVER, "https://fhir.backend.bluebutton.hhsdevcloud.us/baseDstu3");
+		defaultParameters.addArgument(KEYSTORE_DIR, "/opt/fhir_stress/dev/ssl-stores");
+		defaultParameters.addArgument(PROXY_HOST, "null");
+		defaultParameters.addArgument(PROXY_PORT, "0");
+		defaultParameters.addArgument(RIFFILE, "beneficiary_test.rif");
+		defaultParameters.addArgument(RIFTYPE, "BENEFICIARY");
+		return defaultParameters;
+	}
+
+	/**
+	 * @see org.apache.jmeter.protocol.java.sampler.AbstractJavaSamplerClient#setupTest(org.apache.jmeter.protocol.java.sampler.JavaSamplerContext)
+	 */
+	@Override
+	public void setupTest(JavaSamplerContext context) {
+		super.setupTest(context);
+		this.hostName = getHostname();
+
+		if (context.getParameter(PROXY_HOST).equals("null")) { // no proxy
+			this.client = FhirClient.create(context.getParameter(PARAM_SERVER), context.getParameter(KEYSTORE_DIR));
+		} else { // use proxy
+			this.client = FhirClient.create(context.getParameter(PARAM_SERVER), context.getParameter(KEYSTORE_DIR),
+					context.getParameter(PROXY_HOST), Integer.parseInt(context.getParameter(PROXY_PORT)));
```

Comment:
Crap. We **really** don't want our load tests running through that Squid proxy -- it's been flaky and even if it weren't, it's not representative of how real queries come in, and so may throw off our results. Let's leave things as-is for now, but I've filed https://issues.hhsdevcloud.us/browse/CBBI-365 to resolve this once we're running against the non-HHS sandbox (i.e. once we've switched to a GDIT-hosted sandbox).